<html xmlns="http://www.w3.org/1999/xhtml"><head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
    <title>OSCA</title>
    <link type="text/css" rel="stylesheet" href="./css/jquery.tocify.css">
    <link type="text/css" rel="stylesheet" href="./css/bootstrap.min.css">
    <link type="text/css" rel="stylesheet" href="./css/solarized-light.css">
    <link type="text/css" rel="stylesheet" href="./css/main.css">
    <script src="./js/jquery-1.12.4.min.js"></script>
    <script src="./js/jquery-ui.min.js"></script>
    <script src="./js/jquery.tocify.js"></script>
    <script src="./js/hljs.js"></script>
     
</head>

<body>
    <div class="page-header">
        <div class="container-fluid">
            <p class="TOP_head">&nbsp;</p>
            <div class="span6">
                <h1>OSCA</h1>
                <p class="TOP_head">OmicS-data-based Complex trait Analysis</p>
            </div>
            <div class="span6">
                <ul class="nav nav-pills">
                <li role="presentation"><a href="http://cnsgenomics.com/software/gcta/">GCTA</a></li>
                 <li role="presentation"><a href="http://cnsgenomics.com/software/smr/">SMR</a></li>
                  <li role="presentation"><a href="http://cnsgenomics.com/software/gsmr/">GSMR</a></li>
                  <li role="presentation" class="active"><a href="#">OSCA</a></li>
                   <li role="presentation"><a href="http://cnsgenomics.com/software.html">Program in PCTG</a></li>
                    <li role="presentation"><a href="http://gcta.freeforums.net/">CTG forum</a></li>
                </ul>

            </div>
        </div>

    </div>
    <div class="container-fluid">
        <div class="row-fluid">
            <div class="span3">
                <div id="toc"></div>
                <!--[if lt IE 9]>
				<div style=" margin-top: 230px; position: fixed; width: 230px; background-color: yellow; z-index: 10000;">
				<p>Due to our shortage of testing, the document can't view correctly under Internet Explorer <= 8.0. You may need update your browser, or downlad offline <a href="./gcta_doc_latest.pdf">PDF document</a>, <a href="./gcta_latest.zip">executable Files</a></p>
				</div>
				<![endif]-->
            </div>
            <div id="show" class="span9">
                <p> Loading...</p>
                <p> If it not work properly, you may need update your Internet browser and enable javascript</p>
                <p> or downlad offline <a href="./download/OSCA.pdf">PDF document</a>, <a href="./download/OSCA_Linux.zip">executable Files</a></p>
            </div>
            <div id="content" class="span9" style="display: none;"><body><h2 id="overviewexpand" class="expand">Overview </h2>
<hr><h3 id="about">About</h3>
<p> <strong>OSCA</strong> (OmicS-data-based Complex trait Analysis) is a software tool written in C/C++ for the analysis of complex trait using multi-omics data. It is developed by <a href="http://researchers.uq.edu.au/researcher/12709">Futao Zhang</a>,
                        <a href="http://researchers.uq.edu.au/researcher/3051">Zhihong Zhu</a>
                          and <a href="http://researchers.uq.edu.au/researcher/2713">Jian Yang</a> at Institute for Molecular Bioscience,
                        the University of Queensland. Bug reports or questions: Jian Yang <<a href="mailto:jian.yang@uq.edu.au">jian.yang@uq.edu.au</a>>.</p>
                        <p>Functions currently supported are:</p>
                        <ul>
                            <li>
                                <p>Estimating the epigenetic (or transcriptomic) relationships between individuals from genome-wide DNA methylation (or gene expression) data.</p>
                            </li>
                            <li>
                                <p>Estimating the proportion of phenotypic variance for a complex trait can be “explained” by all DNA methylation (or gene expression) probes.</p>
                            </li>
                            <li>
                                <p>Mixed linear model based analysis to test for associations between DNA methylation (or gene expression) probes and a complex trait.</p>
                            </li>
                            <li>
                                <p>Estimating the joint “effects” of all methylation (transcription) probes on a phenotype (e.g. BMI) in a mixed linear model (analogous to BLUP). These estimated effects can be used to predict the phenotype in a new independent sample.</p>
                            </li>
                        </ul>
                        <p><b>Note:</b> Although the software is designed for gene expression and DNA methylation data, it can be applied to any other source of omics data including microbiome, proteome and brain connectome.</p>
                        
<hr><h3 id="credits">Credits</h3>
<p><a href="http://researchers.uq.edu.au/researcher/12709">Futao Zhang</a> and <a href="http://researchers.uq.edu.au/researcher/2713">Jian Yang</a> developed the methods, software and webpage. <a href="http://researchers.uq.edu.au/researcher/3051">Zhihong Zhu</a> contributed to the development of the ORM and OREML methods. <a href="mailto:zhili.zheng@imb.uq.edu.au">Zhili Zheng</a> provided the template of the website.</p>

<hr><h3 id="questionsandhelprequests">Questions and Help Requests</h3>
<p>If you have any bug reports or questions please send an email to <a href="http://scholar.google.com.au/citations?user=aLuqQs8AAAAJ&amp;hl=en">Jian Yang</a> at <a href="mailto:jian.yang@uq.edu.au">jian.yang@uq.edu.au</a></p>
<hr><h3 id="citations">Citations</h3>
<p><strong>EWAS method:</strong><br>
 Zhang, F. et al. (2017) OSCA: a tool for omics-data-based complex trait analysis. In preparation.
                      </p>
<p style="color: rgb(204,0,0);font-weight:bold;">Last update: 21 July 2017</p>
<h2 id="download">Download</h2>
<hr><h3 id="executablefilesnotoc" >Executable Files (version 0.3)  </h3>
 
                    <p>The Linux version is available at: <a href="./download/OSCA_Linux.zip">OSCA_Linux.zip</a>.</p>
                    <p>The MacOS version is available at: <a href="./download/OSCA_Mac.zip">OSCA_Mac.zip</a>.</p>
                    <p>The Windows version is available at: <a href="./download/OSCA_Win.zip">OSCA_Win.zip</a>.</p>
                    <p>The example files are in <a href="./download/example.zip">example.zip</a>.</p>
<p>The executable files (binary code) are release under MIT lincense. </p>
<hr><h3 id="updatelognotoc" >Update log </h3>
<h4 id="version13001july2017">Version 0.30 (21 July 2017)</h4>
<ul>
<li>Beata version released.</li>
</ul>

<h2 id="basicoptions" class="expand">Data Management</h2>

<hr><h3 id="inputandoutput"  >Data Format</h3>
<p>The DNA methylation (or gene expression) data are stored in a binary format for consistency and storage efficiency. We store the data in three separate files .oii (individual information, similar as a PLINK .fam file), .opi (probe information) and .bod (a binary file to store the DNA methylation or gene expression profiles).</p>
 <p><strong><i>myeed.oii</i></strong></p>
                        <pre class="r"><code>
F01	I01	0	0	NA
F02	I01	0	0	NA
F03	I02	0	0	NA
...	</code></pre>
                        <p>Columns are family ID, individual ID, paternal ID, maternal ID and sex (1=male; 2=female; 0=unknown). Missing data are represented by "NA".</p>
                        <p><strong><i>myeed.opi</i></strong></p>
                        <pre><code>
1	probe101	924243	Gene01	+
1	probe102	939564	Gene01	-
1	probe103	1130681	Gene01	-
...	</code></pre>
                        <p>Columns are chromosome, probe ID (can be the ID of an exon or a transcript for RNA-seq data), physical position, gene ID and gene orientation.</p>
                        <p><strong><i>myeed.bod</i></strong></p>
                       <p>DNA methylation (or gene expression) data in binary format.<font color="red"> Please do not try to open this file with a text editor.</font></p>
                       <p> <b>Note:</b> The text below are for advanced users only. We use the first 12 bytes of the .bod file to store the descriptive information for the data. The first 2 bytes are reserved for further extension of the software. The type of data is indicated by the 3rd byte (0 for gene expression data, 1 for DNA methylation data, and 2 for any other type of data). The type of value is indicated by the 4th byte (0 for DNA methylation beta value, 1 for DNA methylation m value and 2 for any other type of value). The number of individuals and number of probes are stored as integers (4 bytes each). For example, for a DNA methylation data set (0x01) of m values (0x01) on 1,342 (0x53e) individuals and 228,694 (0x37d56) CpG sites, the first 12 bytes of the .bod file are: </p>         
                        <pre><code>0101 0000 3e05 0000 567d 0300</code></pre>
                        <p>The bytes afterwards are for the DNA methylation or gene expression data.</p>
                        <p></p>
<hr><h3 id="datamanagement"  >Make a binary file</h3>
 <p>To compile data in binary format</p>
                       
                        <pre class="r"><code> osca --efile myprofile.txt --methylation-beta --make-bod --out myprofile </code></pre>
                         <ul>
                            <li>
                                <p><strong>--efile</strong> reads a DNA methylation (or gene expression) data file in plain text format.</p>
                            </li>
                            <li>
                                <p><strong>--methylation-beta </strong> indicates methylation beta values in the file.</p>
                            </li>
                            <li>
                                <p><strong>--make-bod</strong> saves DNA methylation (or gene expression) data in binary format.</p>
                            </li>
                            <li>
                                <p><strong>--out</strong> saves data ( or results) in a file.</p>
                            </li>
                            
                            </ul>
                            <p><strong><i>myprofile.txt</i></strong></p>
                        <pre><code>FID	IID cg00000658	cg26036652	cg00489772	...
F01	I01	0.909	0.845	0.41	...
F01	I02	0.832	0.732	0.503	...
...
                        </code></pre>
                        <p>This is a file with a header line that contains family ID, individual ID and names of probes. The column of family ID is optional. Please use the flag "--no-fid" for data without family ID.</p>
                        <pre ><code>osca --efile myprofile.txt  --methylation-beta --make-bod --no-fid --out myprofile </code></pre>
                        <ul>
                         <li>
                                <p><strong>--no-fid </strong> indicates data without family ID.</p>
                            </li>
                             </ul>
                        <pre ><code>osca --efile myprofile.txt  --methylation-m --make-bod --out myprofile </code></pre>
                         <ul>
                         <li>
                                <p><strong>--methylation-m </strong> indicates DNA methylation m values in the file.</p>
                            </li>
                             </ul>
                             <p>If the profile is of gene expression value. </p>
                        <pre ><code>osca --efile myprofile.txt  --gene-expression --make-bod --out myprofile </code></pre>
                         <ul>
                         <li>
                                <p><strong>--gene-expression </strong> indicates gene expression profiles in the file.</p>
                            </li>
                             </ul>
                              <p>For any other type of data: </p>
                        <pre ><code>osca --efile myprofile.txt --make-bod --out myprofile </code></pre>
                        <p></p>
                        <p>A binary file also can be made from a transposed file in text format.</p>
                        <pre><code>osca --tefile mytprofile.txt --methylation-beta --make-bod --no-fid --out myprofile</code></pre>
                        <p><strong><i>mytprofile.txt</i></strong></p>
                        <pre><code>FID	F01	F01	...
IID	I01	I02	...
cg00000658	0.909	0.832	...
cg26036652	0.845	0.732	...
cg00489772	0.41	0.503	...
...
                        </code></pre>
                        <p>The row of family ID is optional. Please use the flag "--no-fid" if there is no family ID in your data.</p>
                         <pre ><code>osca --tefile mytprofile.txt --methylation-beta --make-bod --no-fid --out myprofile </code></pre>
                         
<hr><h3 id="multithreadcomputing"  >Update .opi file</h3>
<p>Probe information are sometimes not available in the original DNA methylation (or gene expression) data. These information can be updated the command below.</p>
                          <pre ><code>osca --befile myprofile --update-opi annotated.opi </code></pre>
                          
                          
                         <ul>
                            <li>
                                <p><strong>--befile</strong> reads a DNA methylation (or gene expression) data file in binary format.</p>
                            </li>
                            <li>
                                <p><strong>--update-opi</strong> reads a fully annotated .opi file.</p>
                            </li>                            
                            </ul>
 <hr><h3 id="multithreadcomputing"  >Make a text format file</h3>
 <pre ><code>osca --befile myprofile --make-efile --out myprofile.txt </code></pre>
                           <ul>
                            <li>
                                <p><strong>--make-efile</strong> saves the DNA methylation (or gene expression) data in text format.</p>
                            </li>                          
                            </ul>
                             <pre ><code>osca --befile myprofile --make-tefile --out mytprofile.txt </code></pre>
                           <ul>
                            <li>
                                <p><strong>--make-tefile</strong> saves the DNA methylation (or gene expression) data in transposed text format.</p>
                            </li>                          
                            </ul>
 <hr><h3 id="multithreadcomputing"  >Make a subset of profile data</h3>
 <p>To extract a probe</p>
                             <pre ><code>osca --befile myprofile --probe cg00000658 --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--probe</strong> extracts a specified probe.</p>
                            </li>                          
                            </ul>
                            <p>To exclude a probe</p>
                             <pre ><code>osca --befile myprofile --probe-rm cg00000658 --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--probe-rm</strong> excludes a specified probe.</p>
                            </li>                          
                            </ul>
                            <p>To extract a subset of probes</p>
                             <pre ><code>osca --befile myprofile --extract-probe probe.list --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--extract-probe</strong> extracts a subset of probes.</p>
                            </li>                          
                            </ul>
                             <p><strong><i>probe.list</i></strong></p>
                      <pre><code>cg00000658
cg26036652
...                 </code></pre>
                            <p>To exclude a subset of probes</p>
                             <pre ><code>osca --befile myprofile --exclude-probe probe.list --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--exclude-probe</strong> excludes a subset of probes.</p>
                            </li>                          
                            </ul>
                             <p>To extract a subset of individuals</p>
                             <pre ><code>osca --befile myprofile --keep indi.list --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--keep</strong> extracts a subset of individuals.</p>
                            </li>                          
                            </ul>
                            <p><strong><i>indi.list</i></strong></p>
                      <pre><code>F01	I01
F01	I02
...                  </code></pre>
                            <p>To exclude a subset of individuals</p>
                             <pre ><code>osca --befile myprofile --remove indi.list --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--remove</strong> excludes a subset of individuals.</p>
                            </li>                          
                            </ul>
                            <p>To extract a subset of genes</p>
                             <pre ><code>osca --befile myprofile --genes gene.list --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--genes</strong> extracts a subset of genes.</p>
                            </li>                          
                            </ul>
                            <p><strong><i>gene.list</i></strong></p>
                      <pre><code>MAN1B1
EDEM2
...                    </code></pre>
							<p>To extract a gene</p>
                             <pre ><code>osca --befile myprofile --gene MAN1B1 --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--gene</strong> extracts a gene.</p>
                            </li>                          
                            </ul>
                            <p>To extract a subset of probes based on the order</p>
                             <pre ><code>osca --befile myprofile --from-probe probe0 --to-probe probe1 --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--from-probe</strong> specifies the start probe.</p>
                            </li> 
                            <li>
                                <p><strong>--to-probe</strong> specifies the end probe.</p>
                            </li>                           
                            </ul>
                            <p>To extract a subset of probes in a genomic region centred at a specific probe</p>
                             <pre ><code>osca --befile myprofile --probe cg00000658 --probe-wind 500 --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--probe-wind</strong> defines a window &lt;kb&gt; centred on a specified probe.</p>
                            </li>                          
                            </ul>
                            <p>To extract a subset of probes on a chromosome</p>
                             <pre ><code>osca --befile myprofile --chr 1 --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--chr</strong> specifies a chromosome to select probes.</p>
                            </li>                          
                            </ul>
                              <p>To extract a subset of probes based on physical positions</p>
                             <pre ><code>osca --befile myprofile --from-probe-kb 100 --to-probe-kb 200 --make-bod --out mysubprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--from-probe-kb</strong> specifies the start physical position of the probes.</p>
                            </li> 
                            <li>
                                <p><strong>--to-probe-kb</strong> specifies the end physical position of the probes.</p>
                            </li>                           
                            </ul>
 <hr><h3 id="multithreadcomputing" >Merge binary files</h3>
 <pre ><code>osca --befile-flist mybod.flist --make-bod --out myprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--befile-flist</strong>  reads a file to get the full paths of the binary files.</p>
                            </li>                          
                            </ul>
                            <p><strong><i>mybod.flist</i></strong></p>
                      <pre><code>path1/my_bod1
path2/my_bod2
...                    </code></pre>
						<h4> Methylation beta value to methylation m value and vice versa</h4>
                        <pre ><code>osca --befile myprofile --m2beta --make-bod --out newprofile </code></pre>
<ul>
                            <li>
                                <p><strong>--m2beta</strong> calculates the methylation beta value from the methylation m value.</p>
                            </li>                          
                            </ul>
					<pre ><code>osca --befile myprofile --beta2m --make-bod --out newprofile </code></pre>
							<ul>
                            <li>
                                <p><strong>--beta2m</strong> calculates the methylation m value from the methylation beta value.</p>
                            </li>                          
                            </ul>
 <hr><h3 id="multithreadcomputing" >Calculate variance and mean</h3>
 <pre ><code>osca --befile myprofile --get-variance --get-mean --out newprofile </code></pre>
							<ul>
                            <li>
                                <p><strong>--get-variance</strong> calculates the variance of each probe.</p>
                            </li>   
                            <li>
                                <p><strong>--get-mean</strong> calculates the mean of each probe.</p>
                            </li>                          
                            </ul>
                             <p><strong><i>newprofile.var.txt</i></strong></p>
                      <pre><code>cg00000957	0.000812127
cg00001349	0.00560701
...                 </code></pre>
                    <p><strong><i>newprofile.mean.txt</i></strong></p>
                      <pre><code>cg00000957	0.901574
cg00001349	0.860279
...                 </code></pre> 
 <hr><h3 id="multithreadcomputing" >Data trimming</h3>
 <pre ><code>osca --befile myprofile --std 0.02 --make-bod --out newprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--std</strong> removes the probes with the standard deviation smaller than a specified threshold.</p>
                            </li>                          
                            </ul>
                            <pre ><code>osca --befile myprofile --upper-beta 0.8 --lower-beta 0.2 --make-bod --out newprofile </code></pre>
                            <ul>
                            <li>
                                <p><strong>--upper-beta</strong> removes the DNA methylation probes with the mean beta value larger than a specified threshold.</p>
                            </li> 
                             <li>
                                <p><strong>--lower-beta</strong> removes the DNA methylation probes with the mean beta value smaller than a specified threshold.</p>
                            </li>                             
                            </ul>
 <hr><h3 id="multithreadcomputing" >Quality control with detection p-value</h3>
  <pre ><code>osca --befile myprofile --detection-pval-file dpval.txt --dpval-mth 0  --dpval-thresh 0.05 --make-bod --out newprofile </code></pre>
                                <ul>
                            <li>
                                <p><strong>--detection-pval-file</strong> reads a file that contains DNA methylation detection p-values.</p>
                            </li> 
                             <li>
                                <p><strong>--dpval-mth</strong> specifies a method to do quality control with the detection p-values, 0 for removing the probes with one or more detection p-values violating a threshold (default as 0.05), and 1 for dropping the samples violating a proportion threshold (default as 1%) and simultaneously dropping the probes violating a proportion threshold (default as 1%).</p>
                            </li> 
                             <li>
                                <p><strong>--dpval-thresh</strong> specifies a threshold of detection p-value.</p>
                            </li>                             
                            </ul>
                              <pre ><code>osca --befile myprofile --detection-pval-file dpval.txt --dpval-mth 1  --ratio-probe 0.01 --ratio-sample 0.01 --dpval-thresh 0.05 --make-bod --out newprofile </code></pre>
                               <ul>
                            <li>
                                <p><strong>--ratio-probe</strong> specifies a proportion threshold to remove probes.</p>
                            </li> 
                             <li>
                                <p><strong>--ratio-sample</strong>specifies a proportion threshold to remove individuals.</p>
                            </li>                             
                            </ul>
                              <p>the file "dpval.txt" is in the same format with a transposed profile file. Option "--no-fid" is also valid here.</p>
 <hr><h3 id="multithreadcomputing" >Quality control with missing rate</h3>  
  <pre ><code>osca --befile myprofile --missing-ratio-probe 0.01 --make-bod --out newprofile </code></pre>
                              <ul>
                             <li>
                                <p><strong>--missing-ratio-probe</strong> specifies a missing proportion threshold to remove probes.</p>
                            </li>                             
                            </ul> 
 <hr><h3 id="multithreadcomputing" >Probe pruning</h3> 
  <p style="color:red">To be updated....</p>
 <hr><h3 id="multithreadcomputing" > Probe clumping</h3>  
  <p style="color:red">To be updated....</p>  
 <hr><h3 id="multithreadcomputing" > Estimation of probe correlation structure</h3>  
       <p style="color:red">To be updated....</p>
       
<h2 id="greml">ORM</h2>
<pre><code>osca --befile myprofile --make-orm --out myorm</code></pre>
                        <pre><code>osca --befile myprofile --make-orm-bin --out myorm</code></pre>
                          <ul>
                            <p><strong>--make-orm</strong>
                            <br>or<br>
                                <strong>--make-orm-bin</strong> estimates the omics relationship matrix (ORM) between pairs of individuals from a set of probes and save the lower triangle elements of ORM to binary files. This format is compatible with the <a href="http://cnsgenomics.com/software/gcta/estimate_grm.html">GRM </a> in the <a href="http://cnsgenomics.com/software/gcta/"> GCTA </a> software.</p>
                            </li>                              
                            </ul>
                         <pre><code>osca --befile myprofile --make-orm-gz --out myorm</code></pre>
                          <ul>
                             <li>
                                <p><strong>--make-orm-gz</strong> estimates the ORM and save the lower triangle elements of ORM to compressed plain text files.</p>
                            </li>                              
                            </ul>
                         <pre><code>osca --befile myprofile --make-orm --orm-alg 1 --out myorm</code></pre>
                          <ul>
                             <li>
                                <p><strong>--orm-alg</strong> specifies the algorithm to estimate the ORM. 1 for standardized data of each probe, 2 for centred data of each probe and 3 for standardized data of each individual. The default option is 1.</p>
                            </li>                              
                            </ul>
                            <p>Note that although we describe the options above using DNA methylation and gene expression data, all the options can be applied to any other source of omics data mentioned above.</p>


<h2 id="pca">Principal Component Analysis</h2>
<pre ><code>osca --orm myorm --pca 20 --out mypca</code></pre>
                         <ul>
                             <li>
                                <p><strong>--orm</strong> reads the ORM binary files.</p>
                            </li>   
                             <li>
                                <p><strong>--pca</strong> conducts principal component analysis and saves the first n (default as 20) PCs.</p>
                            </li>                               
                            </ul>
                             <p><strong><i>mypca.eigenval</i></strong></p>
                      <pre><code>122.014
95.3064
70.8055
...                    </code></pre>
                          <p><strong><i>mypca.eigenvec</i></strong></p>
                      <pre><code>R06C01	R06C01	0.012637	-0.00328532	0.0263842	-0.00595246
R05C02	R05C02	0.0106692	-0.0184545	-0.0159826	0.013951
R04C02	R04C02	0.0024727	-0.0118185	-0.0256503	-0.0106235
...                    </code></pre>  
                            <p>Users can also manipulate the ORM in the analysis.</p>
                        <pre><code>osca --orm myorm --keep indi.list --pca 20 --out mypca</code></pre>
                        <pre ><code>osca --orm myorm --remove indi.list --pca 20 --out mypca</code></pre>
                        <pre><code>osca --orm myorm --orm-cutoff 0.05 --pca 20 --out mypca</code></pre>
                        <ul>
                             <li>
                                <p><strong>--orm-cutoff</strong> removes one of a pair of individuals with estimated omics relationships larger than the specified cut-off value.</p>
                            </li>                                
                            </ul>
                            <pre><code>osca --merge-orm myorm.flist --pca 20 --out mypca</code></pre>
                        <ul>
                             <li>
                                <p><strong>--merge-orm</strong> reads multiple ORMs in binary format.</p>
                            </li>                                
                            </ul>
                           <p><strong><i>myorm.flist</i></strong></p>
                      <pre><code>myorm0
myorm1
myorm2
...                    </code></pre> 
<h2 id="gctald">OREML</h2>
<pre><code>osca --reml --pheno my.phen --out myreml</code></pre>
                         <ul>
                             <li>
                                <p><strong>--reml</strong> performs REML (restricted maximum likelihood) analysis. This option is usually followed by the option --orm (one ORM) or --merge-orm (multiple ORMs) to estimate the variance explained by the probes that were used to estimate the omics relationship matrix.</p>
                            </li>  
                            <li>
                                <p><strong>--pheno</strong> reads phenotype data from a plain text file. Missing value should be represented by "NA".</p>
                            </li>                                  
                            </ul>
                             <p><strong><i>my.phen</i></strong></p>
                      <pre><code>R06C01	R06C01	32.6332
R05C02	R05C02	23.9411
R04C02	R04C02	29.7441
...                    </code></pre>  
						<p></p>
 						<pre><code>osca --reml --orm myorm --pheno my.phen --mpheno 1 --out myreml</code></pre>
                        <ul>
                            <li>
                                <p><strong>--mpheno</strong> reads a list of comma-delimited trait numbers if the phenotype file contains more than one trait, e.g. "1,3" tells OSCA to take the first and the third trait for analysis. OSCA always takes the first trait for analysis unless this option is specified.</p>
                                <p style="background-color: #fdd7df"><strong>NOTE:</strong> current version only supports single trait analysis in one run.</p>
                            </li>                                  
                            </ul>
                            
                        <pre><code>osca --reml --orm myorm --pheno my.phen --out myreml</code></pre>
                        <pre><code>osca --reml --orm myorm --pheno my.phen --keep indi.list --out myreml</code></pre>
                        <pre><code>osca --reml --orm myorm --pheno my.phen --orm-cutoff 0.05 --out myreml</code></pre>
                        <p>With multiple ORMs</p>
                        <pre><code>osca --reml --merge-orm myorm.flist --pheno my.phen --out myreml</code></pre>
                        <pre><code>osca --reml --merge-orm myorm.flist --pheno my.phen --reml-alg 0 --out myreml</code></pre>
                        <ul>
                            <li>
                                <p><strong>--reml-alg</strong> specifies the algorithm to do REML iterations, 0 for average information (AI), 1 for Fisher-scoring and 2 for EM. The default option is 0, i.e. AI-REML, if this option is not specified.</p>
                                
                            </li>                                  
                            </ul>
                        <pre><code>osca --reml --merge-orm myorm.flist --pheno my.phen --reml-maxit 100 --out myreml</code></pre>
                        <ul>
                            <li>
                                <p><strong>--reml-maxit</strong> specifies the maximum number of iterations. The default number is 100 if this option is not specified.</p>
                                
                            </li>                                  
                            </ul>
                        <pre><code>osca --reml --orm myorm --pheno my.phen --covar my.covar --out myreml</code></pre>
                        <ul>
                            <li>
                                <p><strong>--covar</strong> reads discrete covariates from a plain text file.</p>
                                
                            </li>                                  
                            </ul>
                             <p><strong><i>my.covar</i></strong></p>
                      <pre><code>R06C01	R06C01	F	0
R05C02	R05C02	F	1
R04C02	R04C02	M	1
...                    </code></pre>  
<pre><code>osca --reml --orm myorm --pheno my.phen --qcovar my.qcovar --out myreml</code></pre>
                        <ul>
                            <li>
                                <p><strong>--qcovar</strong> reads quantitative covariates from a plain text file.</p>
                                
                            </li>                                  
                            </ul>
                             <p><strong><i>my.qcovar</i></strong></p>
                      <pre><code>R06C01	R06C01	25
R05C02	R05C02	16
R04C02	R04C02	30
...                    </code></pre>  

<pre><code>osca --reml --orm myorm --pheno my.phen --reml-est-fix --out myreml</code></pre>
 <ul>
                            <li>
                                <p><strong>--reml-est-fix</strong> displays the estimates of fixed effects on the screen.</p>
                                
                            </li>                                  
                            </ul>
     <pre><code>osca --reml --orm myorm --pheno my.phen --reml-no-lrt --out myreml</code></pre>
 <ul>
                            <li>
                                <p><strong>--reml-no-lrt</strong> turns off the LRT.</p>
                                
                            </li>                                  
                            </ul>                       

<h2 id="ewas">EWAS</h2>

                        <h4>Mixed Linear Model Association</h4>
                        
                        <pre ><code>osca --mlma --befile myprofile --pheno my.phen --out my</code></pre>
                        <p>If you have already computed the ORM</p>
                        <pre ><code>osca --mlma --befile myprofile --pheno my.phen --orm myorm --out my</code></pre>
                        <ul>
                            <li>
                                <p><strong>--mlma</strong> initiates an MLM based association analysis including the target probe (the probe to be tested for association) in the ORM. The results will be saved in a plain text file with <strong>.mlma</strong> as the filename extension.</p>
                                
                            </li>                                  
                            </ul>
                             <p><strong><i>my.mlma</i></strong></p>
                      <pre><code>Chr	Probe	bp	Gene	Orientation	b	se	p
1	cg00003287	201346149	TNNT2	-	-0.156	0.597	0.794
1	cg00008647	207082900	IL24	+	0.032	0.354	0.926
1	cg00009292	50882082 	DMRTA2	-	0.120	1.182	0.919
...                    </code></pre> 
<p>This is a text file with headers. Columns are chromosome, probe, probe BP, gene, orientation, effect size, standard error and p-value. </p>
<p></p>
                        <pre ><code>osca --mlma-loco --befile myprofile --pheno my.phen --out my</code></pre>
                         <ul>
                            <li>
                                <p><strong>--mlma-loco</strong> initiates an MLM based association analysis with the chromosome where the target probe is located excluded from the ORM. The results will be saved in a plain text file with <strong>.loco.mlma</strong> as the filename extension.</p>
                                 </li>                                  
                            </ul>
                            
                                <p></p>
                        <pre ><code>osca --mlma --lxpo 0.1 --befile myprofile --pheno my.phen --out my</code></pre>
                         <pre ><code>osca --mlma-loco --lxpo 0.1 --befile myprofile --pheno my.phen --out my</code></pre>
                         <ul>
                            <li>
                                <p><strong>--lxpo</strong> specifies a percentage of probes to exclude from calculating the ORM. This option should accompany <strong>--mlma</strong> or <strong>--mlma-loco</strong>. It will first conduct a linear-regression-based association analysis and then excludes a percentage of top associated probes from the ORM.</p>
                                
                            </li>                                  
                            </ul>
 <h4>Linear Regression</h4>
                        <pre ><code>osca --befile myprofile --pheno my.phen --linear --out my</code></pre>
                         <pre ><code>osca --befile myprofile --pheno my.phen --qcovar my.qcovar --covar my.covar --linear --out my</code></pre>
                         <ul>
                            <li>
                                <p><strong>--linear</strong> saves linear regression statistics to a plain text file.</p>
                                
                            </li>                                  
                            </ul>
                             <p><strong><i>my.linear</i></strong></p>
                      <pre><code>probeChr	ProbeID	Probe_bp	BETA	SE	P	NMISS
1	cg00003287	201346149	-0.0594	0.531	9.11e-01	1337
1	cg00008647	207082900	0.5003	0.263	5.72e-02	1337
1	cg00009292	50882082	1.0512	1.026	3.06e-01	1337
...                    </code></pre>  
<p>This is a text file with headers. Columns are chromosome, probe, probe BP, effect size, standard error, p-value and number of non-missing individuals. </p>

<h2 id="ewassimu">EWAS simulation</h2>
<p>The phenotypes are simulated based on a set of real DNA methylation (or gene expression) data and a simple model y= sum(x<sub>i</sub>b<sub>i</sub>)+&epsilon;, where y is a vector of phenotypes, x<sub>i</sub> is a vector of raw DNA methylation (or gene expression) profile or standardized profile of the i-th "causal" probe, b<sub>i</sub> is the effect of the i-th causal probe and &epsilon; ~ N(0,var( sum(x<sub>i</sub>b<sub>i</sub>))(1/h<sup>2</sup>-1) ) is a vector of residual effect. </p>
                         <pre ><code>osca --simu-qt --simu-hsq 0.1 --befile myprofile --simu-causal-loci mycausal.list --out mypheno</code></pre>
                          <ul>
                            <li>
                                <p><strong>--simu-qt</strong> smulates a quantitative trait.</p>
                            </li>   
                             <li>
                                <p><strong>--simu-hsq</strong> specifies the proportion of variance in phenotype explained by the causal probes. The default value is 0.1.</p>
                            </li>  
                            <li>
                                <p><strong>--simu-causal-loci</strong> reads a list of probes as causal probes. If the effect sizes are not specified in the file, they will be generated from a standard normal distribution.</p>
                            </li>                                 
                            </ul>
                             <p><strong><i>mycausal.list</i></strong></p>
                      <pre><code>cg04584301	1.55182
cg04839274	-0.106226
cg16648571	0.0257417
...                    </code></pre> 
<p>This is a text file with no headers. Columns are probe ID and effect size. </p>
					<pre ><code>osca --simu-qt --simu-hsq 0.1 --simu-eff-mod 0 --befile myprofile --simu-causal-loci mycausal.list --out mypheno</code></pre>
<ul>
                            <li>
                                <p><strong>--simu-eff-mod</strong> specifies whether or not to standardize the causal probe, 0 for standardized profile and 1 for raw profile. The default value is 0.</p>
                            </li>                                    
                            </ul>
                       <pre ><code>osca --simu-cc 100 300 --simu-hsq 0.1 --simu-k 0.1 --befile myprofile --simu-causal-loci mycausal.list --out mypheno</code></pre>     
                             <ul>
                            <li>
                                <p><strong>--simu-cc</strong> simulates a case-control trait and specifies the number of cases and the number of controls.</p>
                            </li>   
                             <li>
                                <p><strong>--simu-k</strong> specifies the disease prevalence. The default value is 0.1 if this option is not specified.</p>
                            </li>  
                                                        
                            </ul>
<h2 id="prediction">Prediction Analysis</h2>
 <pre><code>osca --reml --orm myorm --pheno my.phen --reml-pred-rand --out myblp</code></pre>
 <ul>
                            <li>
                                <p><strong>--reml-pred-rand</strong> predicts the random effects by the BLUP (best linear unbiased prediction) method. This option is to estimate the aggregated effect of all the probes (used to compute the ORM) to the phenotype of an individual. The aggregated omics effects of all the individuals will be saved in a plain text file <strong>*.indi.blp</strong>.</p>
                                
                            </li>                                  
                            </ul>  
                             <p><strong><i>myblp.indi.blp</i></strong></p>
                      <pre><code>R06C01	R06C0	1.02275	0.07065	1.05692	1.05692
R05C02	R05C02	0.18653	0.27059	0.86650	0.86650
R04C02	R04C02	-0.1982	-0.1673	-0.9209	-0.9209
...                    </code></pre>  
<p>This is a text file with no headers. Columns are family ID, individual ID, an intermediate variable, the aggregated omics effect, another intermediate variable and the residual effect.</p>
                             <pre><code>osca --befile myprofile --blup-probe myblp.indi.blp --out myblp</code></pre>   
                             <ul>
                            <li>
                                <p><strong>--blup-probe</strong> calculates the BLUP solutions for the probe effects.</p>
                                
                            </li>                                  
                            </ul>  
                             <p><strong><i>myblp.probe.blp</i></strong></p>
                      <pre><code>cg04584301	-0.000654646
cg04839274	0.000602484
cg16648571	-4.70356e-05
...                    </code></pre> 
<p>This is a text file with no headers. Columns are probe ID and BLUP of the probe effect. </p>
<p></p>
  <pre><code>osca --befile myprofile --score myblp.probe.blp --out myscore</code></pre> 
  <pre><code>osca --befile myprofile --score myblp.probe.blp 1 2 --out myscore</code></pre> 
  
                             <ul>
                            <li>
                                <p><strong>--score</strong> reads score files for probes and generates predicted omics profiles for individuals. (Note that this option largely follows the --score option in PLINK.) It allows users to specify the column numbers for probe ID and score (the default values are 1 and 2 as shown in the example above). </p>
                                
                            </li>         
                            </ul> 
                   <p><strong><i>myscore.profile</i></strong></p>
                      <pre><code>FID	IID	PHENO	CNT	SCORE
131000028	422572	-9	20000	-7.269198e-07
131000031	243421	-9	20000	9.322096e-06
131000179	338728	-9	20000	-1.250443e-05
...                    </code></pre> 
<p>This is a text file with headers. Columns are family ID, individual ID, phenotype, Number of non-missing probes and score. </p>          
       <pre><code>For example
       In the score file:
cg04584301	-0.065
cg04839274	0.060
cg16648571	-1.03

In the DNA methylation data:
FID	IID	cg04584301	cg04839274	cg16648571
R05C02	R05C02	0.18653	0.27059	0.86650

The score should be:
  ( 0.18653*(-0.065) + 0.27059*0.060 + (-1.03)*0.86650 ) / 3 = (-0.8883841) / 3 = -0.296
                  </code></pre> 
     <b></b>                                                       
  <pre><code>osca --befile myprofile --score myblp.probe.blp --score-has-header --out myscore</code></pre>  
  
    <ul>
                            <li>
                                <p><strong>--score-has-header</strong> indicates probe score file has headers. </p>
                                
                            </li>         
                            </ul>  

<h2 id="reference">Options Reference</h2>
 <table summary="argblock">
                            <tr valign="top">
                                <td><p style="color:#F03">--befile</p></td>
                                <td>
                                    <p>reads a DNA methylation (or gene expression) data file in binary format</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--befile-flist</p></td>
                                <td>
                                    <p>reads a file to get the full paths of the binary files</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--beta2m</p></td>
                                <td>
                                    <p>calculates the methylation m value from the methylation beta value</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--blup-probe</p></td>
                                <td>
                                    <p>calculates the BLUP solutions for the probe effects</p>
                                </td>
                            </tr>
                             <tr valign="top">
                                <td><p style="color:#F03">--chr</p></td>
                                <td>
                                    <p>specifies a chromosome to select probes</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--covar</p></td>
                                <td>
                                    <p>reads discrete covariates from a plain text file</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--detection-pval-file</p></td>
                                <td>
                                    <p>reads a file that contains DNA methylation detection p-values</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--dpval-mth</p></td>
                                <td>
                                    <p>specifies a method to do quality control with the detection p-values</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--dpval-thresh</p></td>
                                <td>
                                    <p>specifies a threshold of detection p-value</p>
                                </td>
                            </tr>
                           <tr valign="top">
                                <td><p style="color:#F03">--efile</p></td>
                                <td>
                                    <p>reads a DNA methylation (or gene expression) data file in plain text format</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--exclude-probe</p></td>
                                <td>
                                    <p>excludes a subset of probes</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--extract-probe</p></td>
                                <td>
                                    <p>extracts a subset of probes</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--from-probe</p></td>
                                <td>
                                    <p>specifies the start probe</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--from-probe-kb</p></td>
                                <td>
                                    <p>specifies the start physical position of the probes</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--gene</p></td>
                                <td>
                                    <p>extracts a gene</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--genes</p></td>
                                <td>
                                    <p>extracts a subset of genes</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--gene-expression</p></td>
                                <td>
                                    <p>indicates gene expression profiles in the file</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--get-mean</p></td>
                                <td>
                                    <p>calculates the mean of each probe</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--get-variance</p></td>
                                <td>
                                    <p>calculates the variance of each probe</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--keep</p></td>
                                <td>
                                    <p>extracts a subset of individuals</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--linear</p></td>
                                <td>
                                    <p>saves linear regression statistics to a plain text file</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--lower-beta</p></td>
                                <td>
                                    <p>removes the DNA methylation probes with the mean beta value smaller than a specified threshold</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--lxpo</p></td>
                                <td>
                                    <p>specifies a percentage of probes to exclude from calculating the ORM</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--m2beta</p></td>
                                <td>
                                    <p>calculates the methylation beta value from the methylation m value</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--make-bod</p></td>
                                <td>
                                    <p>saves DNA methylation (or gene expression) data in binary format</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--make-efile</p></td>
                                <td>
                                    <p>saves the DNA methylation (or gene expression) data in text format</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--make-orm</p></td>
                                <td>
                                    <p>estimates the omics relationship matrix (ORM) and save the lower triangle elements of ORM to binary files</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--make-orm-bin</p></td>
                                <td>
                                    <p>estimates the omics relationship matrix (ORM) and save the lower triangle elements of ORM to binary files</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--make-orm-gz</p></td>
                                <td>
                                    <p>estimates the omics relationship matrix (ORM) and save the lower triangle elements of ORM to compressed plain text files</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--make-tefile</p></td>
                                <td>
                                    <p>saves the DNA methylation (or gene expression) data in transposed text format</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--merge-orm</p></td>
                                <td>
                                    <p>reads multiple ORMs in binary format</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--methylation-m </p></td>
                                <td>
                                    <p>indicates methylation m values in the file</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--methylation-beta</p></td>
                                <td>
                                    <p>indicates methylation beta values in the file</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--mlma</p></td>
                                <td>
                                    <p>initiates an MLM based association analysis including the target probe (the probe to be tested for association) in the ORM</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--mlma-loco</p></td>
                                <td>
                                    <p>initiates an MLM based association analysis with the chromosome where the target probe is located excluded from the ORM</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--mpheno</p></td>
                                <td>
                                    <p>reads a list of comma-delimited trait numbers if the phenotype file contains more than one trait</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--missing-ratio-probe</p></td>
                                <td>
                                    <p>specifies a missing proportion threshold to remove probes</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--no-fid</p></td>
                                <td>
                                    <p>indicates data without family ID</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--orm</p></td>
                                <td>
                                    <p>reads the ORM binary files</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--orm-alg</p></td>
                                <td>
                                    <p>specifies the algorithm to estimate the ORM</p>
                                </td>
                            </tr>
                             <tr valign="top">
                                <td><p style="color:#F03">--orm-cutoff</p></td>
                                <td>
                                    <p>removes one of a pair of individuals with estimated omics relationships larger than the specified cut-off value</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--out</p></td>
                                <td>
                                    <p>saves data (or results) in a file</p>
                                </td>
                            </tr>
                             <tr valign="top">
                                <td><p style="color:#F03">--pca</p></td>
                                <td>
                                    <p>conducts principal component analysis and saves the first n (default as 20) PCs</p>
                                </td>
                            </tr>
                             <tr valign="top">
                                <td><p style="color:#F03">--pheno</p></td>
                                <td>
                                    <p>reads phenotype data from a plain text file</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--probe</p></td>
                                <td>
                                    <p>extracts a specified probe</p>
                                </td>
                            </tr>
                             <tr valign="top">
                                <td><p style="color:#F03">--probe-wind</p></td>
                                <td>
                                    <p>defines a window centred on a specified probe</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--probe-rm</p></td>
                                <td>
                                    <p>excludes a specified probe</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--qcovar</p></td>
                                <td>
                                    <p>reads quantitative covariates from a plain text file</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--ratio-probe</p></td>
                                <td>
                                    <p>specifies a proportion threshold to remove probes</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--ratio-sample</p></td>
                                <td>
                                    <p>specifies a proportion threshold to remove individuals</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--reml</p></td>
                                <td>
                                    <p>performs REML (restricted maximum likelihood) analysis</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--reml-alg</p></td>
                                <td>
                                    <p>specifies the algorithm to do REML iterations</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--reml-est-fix</p></td>
                                <td>
                                    <p>displays the estimates of fixed effects on the screen</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--reml-maxit</p></td>
                                <td>
                                    <p>specifies the maximum number of iterations</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--reml-no-lrt</p></td>
                                <td>
                                    <p>turns off the LRT</p>
                                </td>
                            </tr>
                             <tr valign="top">
                                <td><p style="color:#F03">--reml-pred-rand</p></td>
                                <td>
                                    <p>predicts the random effects by the BLUP (best linear unbiased prediction) method</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--remove</p></td>
                                <td>
                                    <p>excludes a subset of individuals</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--score</p></td>
                                <td>
                                    <p>reads score files for probes and generates predicted omics profiles for individuals</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--score-has-header</p></td>
                                <td>
                                    <p>indicates probe score file has headers</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--simu-causal-loci</p></td>
                                <td>
                                    <p>reads a list of probes as causal probes</p>
                                </td>
                            </tr>
                             <tr valign="top">
                                <td><p style="color:#F03">--simu-cc</p></td>
                                <td>
                                    <p>simulates a case-control trait and specifies the number of cases and the number of controls</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--simu-eff-mod</p></td>
                                <td>
                                    <p>specifies whether or not to standardize the causal probe</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--simu-hsq</p></td>
                                <td>
                                    <p>specifies the proportion of variance in phenotype explained by the causal probes</p>
                                </td>
                            </tr>
                             <tr valign="top">
                                <td><p style="color:#F03">--simu-k</p></td>
                                <td>
                                    <p>specifies the disease prevalence. The default value is 0.1 if this option is not specified</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--simu-qt</p></td>
                                <td>
                                    <p>simulates a quantitative trait</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--std</p></td>
                                <td>
                                    <p>removes the probes with the standard deviation smaller than a specified threshold</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--to-probe</p></td>
                                <td>
                                    <p>specifies the end probe</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--to-probe-kb</p></td>
                                <td>
                                    <p>specifies the end physical position of the probes</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--update-opi</p></td>
                                <td>
                                    <p>reads a fully annotated .opi file</p>
                                </td>
                            </tr>
                            <tr valign="top">
                                <td><p style="color:#F03">--upper-beta</p></td>
                                <td>
                                    <p>removes the DNA methylation probes with the mean beta value larger than a specified threshold</p>
                                </td>
                            </tr>
                        </table>


</body></div>
        </div>

        <script>
            $(function() {
                var toc = $("#toc").tocify({
                    selectors: "h2, h3",
                    ignoreSelector: ".notoc",
                    expand: ".expand",
                    showEffect: "fadeIn",
                    scrollHistory: false,
                    extendPage: false,
                    scrollTo: 80,
                    showDIV: "#show",
                    tocID: "#toc"
                }).data("toc-tocify");
                toc.setOptions({
                    appendContents: '<div style="text-align: center;font-size: 14px">Number of visits: <img src="http://hitwebcounter.com/counter/counter.php?page=6713979&style=0030&nbdigits=5&type=page&initCount=0" title="Home Remedies For Wrinkles" Alt="Home Remedies For Wrinkles"   border="0", width=48 > </div>'
                });
                hljs.initHighlightingOnLoad();

            });
        </script>

    </div>

        <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-101733990-5', 'auto');
  ga('send', 'pageview');

</script>

</body></html>