
## eQTL/mQTL Analysis {: .expand}

This is a module in OSCA to perform a standard eQTL mapping analysis. We describe the module for eQTL mapping but it can be applied to genetic association analysis for all kinds of molecular traits (e.g. DNA methylation, histone modification, metabolites and Microbiome)

### Linear-regression-based

#### \# Basic option 

```
osca --eqtl --bfile mydata --befile myprofile --out myeqtl
```
**\--eqtl** enables the eQTL mapping analysis.

**\--bfile** reads individual-level SNP genotype data (in PLINK binary format), i.e. .bed, .bim, and .fam files.

By default, the results will be saved in [BESD  format ](#BESDformat) . 

#### \# Multi-threading computation
```
osca --eqtl --bfile mydata --befile myprofile --task-num 100 --task-id 1 --thread-num 10 --out myeqtl
```
**\--task-num** specifies the total number of tasks to partition the computation by probes.

**\--task-id** specifies the task IDs form 1 to the total task number.

**\--thread-num** specifies the number of threads for parallel computing. The default value is 1.


#### \# Using other data management options

```
osca --eqtl --bfile mydata --befile myprofile --extract-snp mysnp.list --maf 0.01 --task-num 100 --task-id 1 --thread-num 10 --out myeqtl
```
**\--maf** filters SNPs based on the specified MAF threshold.

#### \# Saving the result in [SMR BESD format](http://cnsgenomics.com/software/smr/#BESDformat).

```
osca --eqtl --bfile mydata --befile myprofile --to-smr --task-num 100 --task-id 1 --thread-num 10 --out myeqtl
```
**\--to-smr** saves the result in SMR compatible format. 

#### \# Fitting covariates

```
osca --eqtl --bfile mydata --befile myprofile --covar mycovar --qcovar myqcovar --task-num 100 --task-id 1 --thread-num 10 --out myeqtl
```
#### \# runing cis-eQTL analysis

```
osca --eqtl --bfile mydata --befile myprofile --cis --task-num 1000 --task-id 1 --thread-num 10 --out myeqtl
```
```
osca --eqtl --bfile mydata --befile myprofile --cis --cis-wind 2000 --task-num 100 --task-id 1 --thread-num 10 --out myeqtl
```
**\--cis** runs a cis-eQTL analysis.

**\--cis-wind** specifies a window (in Kb unit) to store all the SNPs within the window of the probe in either direction. The default value is 2000Kb.

### Mixed-linear-model-based

#### \# runing cis-eQTL analysis
```
osca --eqtl --bfile mydata --befile myprofile --mlm --cis --task-num 100 --task-id 1 --thread-num 10 --out myeqtl
```
**\--mlm** runs a mixed-linear-model-based eQTL analysis on mixed linear model.

```
osca --eqtl --bfile mydata --befile myprofile --mlm --cis --grm mydata --task-num 100 --task-id 1 --thread-num 10 --out myeqtl
```
**\--grm** reads the GRM generated by [GCTA](http://cnsgenomics.com/software/gcta) **\--make-grm** option.
